imap_processing/glows/l0/glows_decom.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+from bitstring import ReadError
+from space_packet_parser import parser, xtcedef
+
+from imap_processing import imap_module_directory
+from imap_processing.glows.l0.glows_l0_data import GlowsDeL0, GlowsHistL0
+
+
+def decom_packets(packet_file_path: str) -> tuple[list[GlowsHistL0], list[GlowsDeL0]]:
+    """Decom GLOWS data packets using GLOWS packet definition
+    Parameters
+    ----------
+    packet_file_path : str
+        Path to data packet path with filename
+
+    Returns
+    -------
+    data : tuple[list[GlowsHistL0], list[GlowsDeL0]]
+        A tuple with two pieces: one list of the GLOWS histogram data, in GlowsHistL0
+        instances, and one list of the GLOWS direct event data, in GlowsDeL0 instance
+    """
+    hist_apid = 1480
+    de_apid = 1481
+
+    # Define paths
+    xtce_document = Path(
+        f"{imap_module_directory}/glows/packet_definitions/GLX_COMBINED.xml"
+    )
+
+    hist_packet_definition = xtcedef.XtcePacketDefinition(xtce_document)
+    histparser = parser.PacketParser(hist_packet_definition)
+
+    histdata = []
+    dedata = []
+
+    with open(packet_file_path, "rb") as binary_data:
+        try:
+            hist_packets = histparser.generator(
+                binary_data,
+                buffer_read_size_bytes=5790778,
+            )
```

Comment:
I didn't use that method because I wanted to use the generator directly and produce the data types as needed, rather than reading and converting the entire packet, converting that to a list, then looping through the list _again_ to process it for glows.

Reply:
I see